Source/Core/VideoCommon/Assets/CustomAssetLoader.h

```diff
@@ -50,14 +50,20 @@ class CustomAssetLoader
                     std::map<CustomAssetLibrary::AssetID, std::weak_ptr<AssetType>>& asset_map,
                     std::shared_ptr<CustomAssetLibrary> library)
   {
+    std::lock_guard lk(m_asset_map_lock);
     auto [it, inserted] = asset_map.try_emplace(asset_id);
     if (!inserted)
       return it->second.lock();
```

Comment:
I'm just noticing this now, but does this actually work correctly for expired weak_ptrs? The weak_ptr doesn't cease to exist, when the pointed at shared_ptr no longer exists, it just returns an empty shared_ptr on lock, right? So I would actually expect something like:

```
    if (!inserted)
    {
      auto shared = it->second.lock();
      if (shared)
        return shared;
    }
```

Reply:
I guess that's what the custom deleter is for, now that I think about it a bit more... but is that completely threadsafe? What if one thread is in the deleter waiting for the `m_asset_map_lock` while another thread is currently in the LoadOrCreateAsset() function? Can that even happen...?

Reply:
When the last shared_ptr gets destroyed, it will clear out the entry from the map, so in theory it should never hit the case of `nullptr` but it's better to be safe.  Ok if I add it to this review?

Reply:
Missed your other comment.  That's a good point, does seem like it could be racey.

Reply:
Does the shared_ptr counts as deleted by the time the deleter is called?? I think I need to debug through this with a test case, this is a very odd construct you've built here.

Reply:
Alternatively, maybe we just accept that dead weak_ptrs can hang around in the map and remove the map-erase in the deleter, that seems a lot more straightforward.

Reply:
Yeah, this approach is a bit of a double edged sword.  Nice from a usability perspective but does have some concerns to think through.  You were right to call this out.  I'm not seeing anything in the std that talks about this.  Will have to look a little closer in a little while.

Reply:
> Alternatively, maybe we just accept that dead weak_ptrs can hang around in the map and remove the map-erase in the deleter, that seems a lot more straightforward.

Yeah, I suppose there aren't going to be that many assets floating around.

Reply:
I decided to remove the map for now.  I'm going to investigate this on the side to see if we can still have the auto delete